setup.py

```diff
@@ -124,17 +124,26 @@ def get_output(*args, **kwargs):
 
 # add in the extra workarounds for different include structures
 if winkrb_path:
-    prefix = winkrb_path
+    prefixes = [winkrb_path]
 else:
+    prefixes = []
     try:
-        prefix = get_output(f"{kc} gssapi --prefix")
+        prefixes.append(get_output(f"{kc} gssapi --prefix"))
     except Exception:
-        print("WARNING: couldn't find krb5-config; assuming prefix of %s"
-              % str(sys.prefix))
-        prefix = sys.prefix
-gssapi_ext_h = os.path.join(prefix, 'include/gssapi/gssapi_ext.h')
-if os.path.exists(gssapi_ext_h):
-    compile_args.append("-DHAS_GSSAPI_EXT_H")
+        print("WARNING: couldn't find krb5-config to determine prefix")
+    try:
+        prefixes.append(os.path.dirname(
+            get_output("pkg-config --variable=includedir krb5-gssapi"))
```

Comment:
I don't know too much about `pkg-config` but is `krb5-gssapi` unique enough across the platforms? What governs this id and are we always sure it's going to match krb5/gssapi. It also brings into question how does it deal with Heimdal.

It should also probably deal with an empty string being outputted as well.

Reply:
It works with both packages on NixOS, and if it is missing it returns and empty string to stdout. I don't have any other machines at this time I could test on.

```
$ nix-shell -p heimdal.dev pkg-config --run "pkg-config --variable=includedir krb5-gssapi"
/nix/store/6w2xlb77p6y8p3amb7q1kw3pbdm4kkmx-heimdal-7.8.0-dev/include

$ nix-shell -p krb5.dev pkg-config --run "pkg-config --variable=includedir krb5-gssapi"
/nix/store/hwv4s9dxmwg7ahzjzlarhwjkirvwzlyb-krb5-1.20.1-dev/include

$ nix-shell -p pkg-config --run "pkg-config --variable=includedir krb5-gssapi"
Package krb5-gssapi was not found in the pkg-config search path.
Perhaps you should add the directory containing `krb5-gssapi.pc'
to the PKG_CONFIG_PATH environment variable
No package 'krb5-gssapi' found
```

The name is governed by the pkg-config file

```
$ cat /nix/store/lg22dp1sij739swldvrf7410r2j209l0-heimdal-7.8.0-dev/lib/pkgconfig/krb5-gssapi.pc
prefix=/nix/store/cy2j8bfp8dckdxaf2k9m0d3k4mqrpgrl-heimdal-7.8.0
exec_prefix=/nix/store/cy2j8bfp8dckdxaf2k9m0d3k4mqrpgrl-heimdal-7.8.0
libdir=/nix/store/cy2j8bfp8dckdxaf2k9m0d3k4mqrpgrl-heimdal-7.8.0/lib
includedir=/nix/store/lg22dp1sij739swldvrf7410r2j209l0-heimdal-7.8.0-dev/include
vendor=Heimdal

Name: krb5-gssapi
Description: Kerberos implementation of the GSS API.
Version: 7.8.0
Requires: heimdal-gssapi

$ cat /nix/store/y9rk7j5afwwzsjk6ximhipngbap8v8vg-krb5-1.20.1-dev/lib/pkgconfig/krb5-gssapi.pc
prefix=/nix/store/nnkw7wdy3f5nlw912hksc82n15s7cwkd-krb5-1.20.1
exec_prefix=${prefix}
libdir=/nix/store/nnkw7wdy3f5nlw912hksc82n15s7cwkd-krb5-1.20.1/lib
includedir=/nix/store/y9rk7j5afwwzsjk6ximhipngbap8v8vg-krb5-1.20.1-dev/include
vendor=MIT

Name: krb5-gssapi
Description: Kerberos implementation of the GSSAPI
Version: 1.20.1
Requires: mit-krb5-gssapi
```

Reply:
> It should also probably deal with an empty string being outputted as well.

In case pkg-config does not find anything, it exits with 1. I can catch `CalledProcessError` and that should be fine, no?

Reply:
> The name is governed by the pkg-config file

But what is setting that name, is it the package maintainers, do they all state that this package should be `krb5-gssapi` or are there distros where that is not the case.

> In case pkg-config is unavailable or does not find anything, it exits with 1. I can catch CalledProcessError and that should be fine, no?

I'm talking about if `pkg-config` is available but the configure value is an empty string. I have no idea if this is a valid scenario but by adding an empty string the header check will check in `include/gssapi/...` relative to the current dir rather than it being rooted somewhere.

Reply:
> But what is setting that name, is it the package maintainers, do they all state that this package should be krb5-gssapi or are there distros where that is not the case.

The names are upstream-maintained.

https://github.com/heimdal/heimdal/blob/f62e2f278437ff6c03d2d09bd628381c795bba78/tools/krb5-gssapi.pc.in#L7
https://github.com/krb5/krb5/blob/0f870b1bcad960fd5319a3f97aafd7f4a289e2fb/src/build-tools/krb5-gssapi.pc.in#L7

> I'm talking about if pkg-config is available but the configure value is an empty string. I have no idea if this is a valid scenario but by adding an empty string the header check will check in include/gssapi/... relative to the current dir rather than it being rooted somewhere.

Given that these files are maintained by upstream, the risk of running into empty values should be negligible.